### PR TITLE
Feature/icsaas 380 improve autocomplete

### PR DESCRIPTION
**Pull request by kevinolivar**
- fix flickering (suggestions html element was re-created each time upon request)
- bold query string in results
- remove border bottom between suggestions except last element
- trigger search upon selecting a suggestion

### CODE REVIEW

**Review by nhirrle — APPROVED** [review]

**Comment by nhirrle** [comment]
could you leave a screenshot / gif?

**Comment by kevinolivar** [comment]
**Before on the left, with improvement on the right:**
https://user-images.githubusercontent.com/81755864/150835451-0237de21-4273-41b0-994b-5b60a9b59d14.mov

**Styling fixes:**
Capture d’écran 2022-01-24 à 18 41 17

**Review by nhirrle — APPROVED** [review]

File: ui.frontend/src/main/webpack/site/js/components/searchInput.ts [hunk 1]
```diff
@@ -45,19 +47,39 @@ const debouncedSearch = (autoSuggestionDebounceTime: number) =>
...
+        removeSuggestionList(searchContainer)
+      }
+
       if (query.length >= autocompleteTriggerThreshold) {
+        const regexp = new RegExp(query, 'gi')
+        const searchButtonElement = document.getElementsByClassName(
+          'saas-container_button',
+        )?.[0] as HTMLElement
         const results = await fetchAutoComplete(autocompleteUrl, query)
+        let suggestionDropdown: any = null
```
Comment (jerome-mottet-valtech) [inline ui.frontend/src/main/webpack/site/js/components/searchInput.ts:62]: isn't the type a `HTMLElement` ? Or div element?

**Review by jerome-mottet-valtech — APPROVED** [review]

File: ui.frontend/src/main/webpack/site/js/components/searchInput.ts (hunk 1 above)
Comment (kevinolivar) [reply ui.frontend/src/main/webpack/site/js/components/searchInput.ts:62]: yes and it complains with null :D

Maybe I can do HTMLElement | null

Comment (jerome-mottet-valtech) [reply ui.frontend/src/main/webpack/site/js/components/searchInput.ts:62]: That should work 👍

**Comment by kevinolivar** [comment]
@nhirrle 

a little improvement with trimming.

Capture d’écran 2022-01-25 à 07 48 09

**Pull request merged**